newIDE/app/src/UI/SemiControlledTextField.js

```diff
@@ -121,7 +144,17 @@ export default class SemiControlledTextField extends React.Component<
           if (!commitOnBlur) onChange(newValue);
         }}
         onBlur={event => {
-          onChange(event.currentTarget.value);
+          let value = event.currentTarget.value;
+          if (this.props.calculValue) {
+            const rex = /(^\d).*[-+*()/.,].*(\d+$)|(^\d+$)|^((\d)|[-+*()/.,]).((\d)|[-+*()/.,])+$/gm;
+            const regexData = value.match(rex);
+
+            if (regexData) {
+              value = eval(regexData[0]);
```

Comment:
I still think it would be simpler and probably more effective to just use a module made for evaluating mathematical expressions, like https://github.com/josdejong/mathjs-expression-parser

Reply:
I like use vanilla javascript, systematically add a package for solve a problem I always found it ridiculous and overrated. [I saw nice thread](https://stackoverflow.com/questions/7650071/is-there-a-way-to-create-a-function-from-a-string-with-javascript).

Reply:
I do agree that using a package for everything is not a good solution, there are also packages made [to make fun of this](https://github.com/jezen/is-thirteen), but I don't think modules should be the very last resort either. I think this case is a good fit for an external module, as the eval is made to evaluate code, not maths, and using it for maths is missuing it. 

Your regex is not completely preventing javascript, this input for example matches your regex: `20;alert("some malicious code");2+3`. I doubt any regex can be 100% secure. Someone could make a really long expression containing this kind of malicious code, and tell a user to paste it in for a desired result, executing malicious code on his machine.

The only way to really not execute code, is not executing that as code in the first place. Simply parse it as a mathematical expression, what it is, and execute it. Of course, something like a math expression parser is very long and difficult to do. Fortunately, someone made one already. Why spend time remaking the wheel, when we can just use the code from a module?

Reply:
Vanilla JavaScript is most of the time the way to go, but here for `eval` I would consider it almost not part of JS because of all the implications it has.

Taking a new dependency comes with a lot of factor that are both in favor or against it:
- size of the dependency?
- complexity of the dependency?
- is this something critical, or just something that we don't have time for and we're happy to delegate?
- any security risk with (or without the dependency)?
- can we just replace it with a few lines of code?
- any performance issue?
- is the package going to be maintained for the next years? If not do we have the capacity to take over it?

Here there are two things:
- I really don't want eval because there will be one way or the other a chance to inject code. It's also something that we could disable completely thanks to a [CSP directive](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) - so let's not build anything that relies on it. There are ways to do complex code in JS with as little as simple characters like [, ] and +.
  - I really don't want it :p It's sure it will be abused in the future. Or we'll get angry users yelling at us. Or both.
- I also don't want to get a complex math parser just for a simple, convenient feature like this. So can we find a dead simple, very lightweight, minimal math parser?
  - this one has no dependencies: https://github.com/bugwheels94/math-expression-evaluator
  - this one too? https://github.com/silentmatt/expr-eval/blob/master/package.json